### DISCUSSION TITLE
Does Borg delete old backups?

### DISCUSSION THREAD

**Discussion by SamuXzX** [nextcloud/all-in-one#886] [Questions] [answered]
I enabled daily backups but I'm wondering if I will soon find my server full of backups. Is there a limit number of backups Borg keeps or do I have to delete them manually?

**Comment by szaimen** [comment, marked as answer]
> Does Borg delete old backups? 

Yes, it does

> I enabled daily backups but I'm wondering if I will soon find my server full of backups.

No this will not happen if you have enough storage. The backup archive shoild not take more space than the backed up data.

**Reply by SamuXzX** [reply]
I'm sorry but I think we are missing informations in the documentation. In my Nextcloud AIO dashboard I see 8 backups from the last 8 days. How much old a backup needs to be to be deleted? What does "The backup archive should not take more space than the backed up data" mean if Nextcloud AIO is storing multiple backups (so that together they will be larger than the backed up data)?

Given the line @dantefromhell found:
```
# Prune options
    BORG_PRUNE_OPTS=(--stats --progress --keep-within=7d --keep-weekly=4 --keep-monthly=6 "$BORG_BACKUP_DIRECTORY")
``` 
Does it mean that it will keep 6 backup per month, 4 per week and only the backups of the last 7 days? Isn't this last rule overwriting the other two?

**Reply by szaimen** [reply]
You should read https://borgbackup.readthedocs.io/en/stable/usage/prune.html and https://github.com/borgbackup/borg#what-is-borgbackup and below

**Reply by SamuXzX** [reply]
Ok I see. The rules are not  progressive filters: they individually [add archives to keep](https://borgbackup.readthedocs.io/en/stable/usage/prune.html#description), and Borg [only keeps changes](https://github.com/borgbackup/borg#main-features) so that it effectively never exceeds the size of the backed up data.

Thank you.

**Comment by dantefromhell** [comment]
It is this section that defines backup retention policy, correct?

https://github.com/nextcloud/all-in-one/blob/3de963c238b77c1ba93ac92e614d836b8e046b48/Containers/borgbackup/backupscript.sh#L142

**Reply by szaimen** [reply]
Yes

**Reply by dantefromhell** [reply]
A great feature (non-urgent from my perspective) would be to allow aio admins to configure the retention periods.
